src/iop/watermark.c

```diff
@@ -926,12 +1012,14 @@ static void load_watermarks(const char *basedir, dt_iop_watermark_gui_data_t *g)
   for(GList *iter = files; iter; iter = g_list_next(iter))
   {
     char *filename = iter->data;
-    // remember the whole filename for later
-    g->watermarks_filenames = g_list_append(g->watermarks_filenames, g_strdup(filename));
-    // ... and remove the file extension from the string shown in the gui
-    char *c = strrchr(filename, '.');
-    if(c) *c = '\0';
-    dt_bauhaus_combobox_add(g->watermarks, filename);
+    const gchar *extension = strrchr(filename, '.');
```

Comment:
again shouldn't we guard against extension being NULL ?

Reply:
done